components/edit_channel_header_modal/edit_channel_header_modal.jsx

```diff
@@ -68,6 +68,15 @@ class EditChannelHeaderModal extends React.PureComponent {
         }).isRequired,
     }
 
+    static getDerivedStateFromProps(props) {
+        const {requestStatus} = props;
+        if (requestStatus === RequestStatus.SUCCESS) {
+            return {show: false, showError: false};
+        }
+
+        return {showError: requestStatus === RequestStatus.FAILURE};
```

Comment:
Could you remove `this.state.showError`? In the render method, we can just control the error text by whether or not `this.props.serverError` is set